.github/workflows/cd-sql-engine-populate-persistent-source-schema.yaml

```diff
@@ -0,0 +1,106 @@
+# See [Persistent Source Schema](/GLOSSARY.md#persistent-source-schema)
+# Populating the source schema via this workflow ensures that it's done with the same settings as the tests.
+
+name: Reload Test Data in SQL Engines
+
+# We don't want multiple workflows trying to create the same table.
+concurrency:
+  group: POPULATE_PERSISTENT_SOURCE_SCHEMA
+  cancel-in-progress: true
+
+on:
+  pull_request:
+    types: [labeled]
+
+env:
+  # Unclear on how to make 'Reload Test Data in SQL Engines' a constant here as it does not work here.
+  PYTHON_VERSION: "3.8"
+  ADDITIONAL_PYTEST_OPTIONS: "--use-persistent-source-schema"
+
+jobs:
+  snowflake-populate:
+    environment: DW_INTEGRATION_TESTS
+    if: github.event.action == 'labeled' && github.event.label.name == 'Reload Test Data in SQL Engines'
```

Comment:
This may need an update for workflow_dispatch to fire, but it should be simple enough.